addons/udes_stock/models/stock_move.py

```diff
@@ -1,10 +1,56 @@
 # -*- coding: utf-8 -*-
-from odoo import models, _
+from odoo import models, api, _
 
 
 class StockMove(models.Model):
     _inherit = "stock.move"
 
+    def _prepare_move_line(self, moves, qty, **kwargs):
```

Comment:
Only one qty for a list of moves? shouldn't this be a list of quantities with the same length as moves recordset?

Reply:
Yeah good point, is way too restrictive with just the one quantity for n moves. Is a list of quantities safe enough though (if safe is the right word)? We could just have a ```moves_info``` dict with the move as the key and the quantity as the value.

Reply:
We'd definitely want to be able link the quantity back to the required move, although we'd still want to make sure that any calls of `_prepare_move_line` are not overly complicated. Perhaps `_prepare_move_line` and `_prepare_move_lines`?

Reply:
Yeah think that makes sense. That way you have a quick option for a single move line or a slightly more complex way (but still far quicker than creating move lines manually) of creating a batch of move lines.

Reply:
Have now added a new `_prepare_move_lines` helper method which itself calls `_prepare_move_line` for each move and quantity supplied.

Reply:
Personally I'd prefer a single function that can accept a dictionary or a list of dictionaries similar to how odoo13 does .create()